### PR TITLE
Nikita/farm token fix

### PR DESCRIPTION
**Pull request by nkrulikovsky**
## Farm token fix.
Farm token doesn't display because in some case pull name from gauge is null.

It fix this issue by getting poolTokens from basicPoolContext.

But the following 3 gauges don't have pool addresss.

`saddleTBTC-V3-gauge`
`saddleWCUSD-V3-gauge`
`saddleUSDCUSDTFrax-gauge`

### CODE REVIEW

File: src/pages/Farm/FarmOverview.tsx
Comment (ug02fast) [inline src/pages/Farm/FarmOverview.tsx:65]: If it makes sense, instead of returning <div></div> can you `.filter` with same condition before the `.map`?

File: src/pages/Farm/FarmOverview.tsx
Comment (ug02fast) [inline src/pages/Farm/FarmOverview.tsx:70]: Can you change these to be of a similar pattern?

File: src/pages/Farm/FarmOverview.tsx
Comment (ug02fast) [inline src/pages/Farm/FarmOverview.tsx:63]: Possible move and handle this `!chainId` check higher up? Outside of the loop?

**Review by ug02fast — CHANGES_REQUESTED** [review]

**Review by hammeiam — APPROVED** [review]
Not ideal, but it works! We'll need to remove getTokenByAddress in the future. I'm still not sure why usePoolData wasn't working for this

File: src/pages/Farm/FarmOverview.tsx
Comment (nkrulikovsky) [reply src/pages/Farm/FarmOverview.tsx:65]: Because I think we don't able to use filter because the poolTokens is only address of token, not token.

File: src/pages/Farm/FarmOverview.tsx
Comment (nkrulikovsky) [reply src/pages/Farm/FarmOverview.tsx]: good call out!

File: src/constants/index.ts
Comment (hammeiam) [inline src/constants/index.ts:1862]: DO NOT MERGE THIS

**Review by hammeiam — CHANGES_REQUESTED** [review]

**Review by hammeiam — APPROVED** [review]

**Review by ug02fast — APPROVED** [review]
lgtm, you can also return `null` if chainId falsy

File: src/pages/Farm/FarmOverview.tsx
Comment (ug02fast) [reply src/pages/Farm/FarmOverview.tsx:65]:
```
poolTokens?.filter((tokenAddress) => Boolean(getTokenByAddress(tokenAddress, chainId)).map((tokenAddress) =>  
{
  const token = getTokenByAddress(...
 ```
 
Something like this should work but you have to double-check if you want to do this. After filtering you will only have valid tokens so there will be no use for `if (!token)` case.

File: src/pages/Farm/FarmOverview.tsx
Comment (ug02fast) [inline src/pages/Farm/FarmOverview.tsx:41]: Can you add extra newline after this for style purposes? You can also return `null`

Comment (nkrulikovsky) [reply src/pages/Farm/FarmOverview.tsx:41]: BTW, getTokenByAddress is return `Token||null` so after filtering we will get null again so we need to check again.

**Pull request merged**
